### PR TITLE
Tidy up local development tooling

### PR DESCRIPTION
**Pull request by nottrobin**
- Remove JavaScript directories and tooling (not used anywhere in the site)
- Remove redundant gulp functions
- Remove unused NPM dependencies
- Define "clean" endpoint so `./run clean` will delete everything it should

QA
--

- Read the code
- Check `./run clean` and `./run` still work

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,14 +4,10 @@
   "description": "Codebase for https://vanillaframework.io",
   "main": "_site/index.html",
   "scripts": {
-    "help": "gulp help",
-    "gulp": "gulp",
-    "browser-sync": "gulp browser-sync",
     "build": "gulp build",
```
Comment (anthonydillon) [inline package.json]: Didn't we agree it would be `serve` for websites?

**Comment by anthonydillon** [comment]
QA is fine, just the question on build over serve.

File: package.json (hunk 1 above)
Comment (anthonydillon) [inline package.json]: Infact I see its build in the run script.

Comment (nottrobin) [inline package.json]: Yep. In this project, Jekyll does the serving, not node. So node doesn't worry itself about `serve`, because it's handled outside.

**Pull request merged**